src/components/badgedLink/BadgedLink.js

```diff
@@ -0,0 +1,105 @@
+import uiTypography from '@teamleader/ui-typography';
+import uiUtilities from '@teamleader/ui-utilities';
+import cx from 'classnames';
+import PropTypes from 'prop-types';
+import React, { forwardRef, useImperativeHandle, useRef } from 'react';
+import Box from '../box';
+import theme from './theme.css';
+
+const BadgedLink = forwardRef(
+  (
+    {
+      children,
+      className,
+      disabled,
+      icon,
+      iconPlacement,
+      element,
+      inherit,
+      inverse,
+      selected,
+      onMouseUp,
+      onMouseLeave,
+      ...others
+    },
+    ref,
+  ) => {
+    const badgedLinkRef = useRef();
+    useImperativeHandle(ref, () => badgedLinkRef.current);
+
+    const blur = () => {
+      const currentBadgedLinkRef = badgedLinkRef.current;
+      if (currentBadgedLinkRef.blur) {
+        currentBadgedLinkRef.blur();
+      }
+    };
+
+    const handleMouseUp = (event) => {
+      blur();
+      onMouseUp && onMouseUp(event);
+    };
+
+    const handleMouseLeave = (event) => {
+      blur();
+      onMouseLeave && onMouseLeave(event);
+    };
+
+    const classNames = cx(
+      uiUtilities['reset-font-smoothing'],
+      uiTypography['ui-text-body'],
+      theme['badged-link'],
+      {
+        [theme['is-inherit']]: inherit,
+        // Since children can be text node, we have to do spacing according to icon element
+        [theme['is-left-icon-placed']]: iconPlacement === 'left',
+      },
+      className,
+    );
+
+    return (
+      <Box
+        element={element}
+        ref={badgedLinkRef}
+        className={classNames}
+        data-teamleader-ui="badgedLink"
+        onMouseUp={handleMouseUp}
+        onMouseLeave={handleMouseLeave}
+        {...others}
+      >
+        {icon && iconPlacement === 'left' && icon}
+        {children}
+        {icon && iconPlacement === 'right' && icon}
```

Comment:
If the icon is actually required like the proptypes say

```suggestion
        {iconPlacement === 'left' && icon}
        {children}
        {iconPlacement === 'right' && icon}
```